301-jenkins-vmss-blue-green/azuredeploy.json

```diff
@@ -0,0 +1,615 @@
+        },
+        "virtualMachineSize": {
+            "type": "string",
+            "defaultValue": "Standard_DS2_v2",
+            "allowedValues": [
+                "Standard_A2_v2",
+                "Standard_A2",
+                "Standard_B2s",
+                "Standard_B2ms",
+                "Standard_D2s_v3",
+                "Standard_D2_v3",
+                "Standard_DS2_v2",
+                "Standard_D2_v2",
+                "Standard_DS2",
+                "Standard_D2"
+            ],
+            "metadata": {
+                "description": "The virutal machine size to use. We picked out the sizes with 2 vCPUs, but in real world projects you can choose other sizes as you desired."
+            }
+        },
+        "jenkinsDNSPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique DNS name prefix for the public IP used to access the Jenkins service."
+            }
+        },
+        "tomcatDNSPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique DNS name prefix for the public IP used to access the Tomcat service."
+            }
+        },
+        "servicePrincipalAppId": {
+            "type": "string",
+            "metadata": {
+                "description": "Service Principal App ID (also called Client ID) that has contributor rights to the subscription used for this deployment. It is used to manage related Azure resources, e.g., bake OS image, check resource status, update VMSS image, etc."
+            }
+        },
+        "servicePrincipalAppKey": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Service Principal App Key (also called Client Secret) that has contributor rights to the subscription used for this deployment. It is used to manage related Azure resources, e.g., bake OS image, check resource status, update VMSS image, etc."
+            }
+        },
+        "_artifactsLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+            },
+            "defaultValue": "https://raw.githubusercontent.com/Azure/jenkins/master"
+        },
+        "_artifactsLocationSasToken": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+            },
+            "defaultValue": ""
+        }
+    },
+    "variables": {
+        "service": "tomcat",
+        "baseImageName": "[concat(parameters('resourceNamePrefix'), '-base-', uniqueString(subscription().id))]",
```

Comment:
We cannot update VMSS with an image in a different region. The image has to be in the same subscription and same region as the target VMSS. We need to copy the image to the target region if we need to use the same image in different regions.

Reply:
So if I run this deployment twice in the same region and in the same subscription - this will result in a naming collision?  It sounds like we just need to seed this way: uniqueString(resourceGroup().id)